actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -75,18 +73,22 @@ def formats
 
       # Sets the \variant for template.
       def variant=(variant)
-        if variant.is_a?(Symbol)
-          @variant = [variant]
-        elsif variant.nil? || variant.is_a?(Array) && variant.any? && variant.all?{ |v| v.is_a?(Symbol) }
-          @variant = variant
+        variant = Array(variant)
+
+        if variant.all? { |v| v.is_a?(Symbol) }
+          @variant = VariantInquirer.new(variant)
         else
-          raise ArgumentError, "request.variant must be set to a Symbol or an Array of Symbols, not a #{variant.class}. " \
+          raise ArgumentError, "request.variant must be set to a Symbol or an Array of Symbols. " \
```

Comment:
I'll put it back. It seemed awkward that if you set `request.variant` to an array of non-Symbol objects, you'd see:

> request.variant must be set to a Symbol or an Array of Symbols, not a Array.